src/vehicle.cpp

```diff
@@ -3752,9 +3762,52 @@ void vehicle::idle(bool on_map) {
 
     if( on_map ) {
         update_time();
+        if(scoop_on){
+            operate_scoop();
+        }
+    }
+}
+void vehicle::operate_scoop()
+{
+    std::vector<int> scoops = all_parts_with_feature( "SCOOP" );
+    auto veh_points = get_points();
+    for( int scoop : scoops ) {
+        const int chance_to_damage_item = 9;
+        int max_pickup_size = parts[scoop].info().size/10;
+        const char *sound_msgs[] = {_("Whirrrr"), _("Ker-chunk"), _("Swish"), _("Cugugugugug")};
+        sounds::sound( global_pos3() + parts[scoop].precalc[0], rng( 20, 35 ), sound_msgs[rng( 0, 3 )] );
+        for( const tripoint &position : veh_points ) {
```

Comment:
Eh, I disagree personally.
 I also preferred when it would stick items anywhere, but if it gets it added to the game, I guess I'll just crush my dream of a convenient part, perfect for cleaning up after parties and zombie apocalypses.

 Well I guess that it will do just fine, I mean, the changes will increase the clean up time significantly, but I suppose that if the survivor can clean too quickly, they might actually get out of the immense mess caused by mankind's hubris. We can't have that now, can we?